src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOUR_INDEX = 2;
+    private static final int INCOME_INDEX = 3;
+    private static final int DAY_BEFORE_OR_AFTER = 1;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate beginningDate = LocalDate.parse(dateFrom, FORMATTER);
+        LocalDate endingDate = LocalDate.parse(dateTo, FORMATTER);
+        StringBuilder result = new StringBuilder();
+        result.append("Report for period ")
+                .append(dateFrom)
+                .append(" - ")
+                .append(dateTo);
+        String endResult = calculateSalary(names, data, beginningDate, endingDate, result);
+        return endResult;
+    }
+
+    public String calculateSalary(String[] names, String[] data, LocalDate beginningDate,
+                                  LocalDate endingDate, StringBuilder result) {
+        for (String name : names) {
+            int salary = 0;
+            for (String dataString : data) {
+                String[] dataSplitArray = dataString.split(" ");
+                LocalDate date = LocalDate.parse(dataSplitArray[DATE_INDEX], FORMATTER);
+                if (date.isAfter(beginningDate.minusDays(DAY_BEFORE_OR_AFTER))
```

Comment:
Not fixed.
1. You were supposed to extract this condition to a separate method.
2. Date check is still over-complicated. You can achieve the simple solution with exclusion/negation:
`!date.isBefore(dateFrom) && !date.isAfter(dateTo);`